tests/Annotations/Command/QueueWatchCommandTest.php

```diff
@@ -116,6 +119,70 @@ public function it_retries_failures()
         $this->assertSame(1, $this->queue->count());
     }
 
+    /**
+     * @test
+     */
+    public function it_will_remove_queue_item_if_upsert_fails()
+    {
+        $item = new InternalSqsMessage('profile', 'username');
+        $profile = new Profile(
+            'username',
+            new PersonDetails('PreferredName', 'IndexName'),
+            new EmptySequence(),
+            new EmptySequence()
+        );
+        $data = [
+            'authority' => $this->authority,
+            'username' => 'username',
+            'email' => 'username@blackhole.elifesciences.org',
+            'display_name' => 'PreferredName',
+        ];
+        $this->transformer
+            ->expects($this->once())
+            ->method('transform')
+            ->with($item)
+            ->will($this->returnValue($profile));
+        $post_request = new Request(
+            'POST',
+            'users',
+            ['Authorization' => $this->authorization, 'User-Agent' => 'HypothesisClient'],
+            json_encode($data)
+        );
+        $post_response_mess = json_encode(['status' => 'failure', 'reason' => 'user with username username already exists.']);
+        $post_response = new Response(400, [], $post_response_mess);
+        $rejected_post_response = new RejectedPromise(new BadResponse($post_response_mess, $post_request, $post_response));
+        $this->httpClient
+            ->expects($this->at(0))
+            ->method('send')
+            ->with(RequestConstraint::equalTo($post_request))
+            ->willReturn($rejected_post_response);
+        $patch_data = [
+            'email' => 'username@blackhole.elifesciences.org',
+            'display_name' => 'PreferredName',
+        ];
+        $patch_request = new Request(
+            'PATCH',
+            'users/username',
+            ['Authorization' => $this->authorization, 'User-Agent' => 'HypothesisClient'],
+            json_encode($patch_data)
+        );
+        $patch_response_mess = json_encode(['status' => 'failure', 'reason' => 'Either the resource you requested doesn\'t exist, or you are not currently authorized to see it.']);
+        $patch_response = new Response(404, [], $patch_response_mess);
+        $rejected_patch_response = new RejectedPromise(new BadResponse($patch_response_mess, $patch_request, $patch_response));
+        $this->httpClient
+            ->expects($this->at(1))
+            ->method('send')
+            ->with(RequestConstraint::equalTo($patch_request))
```

Comment:
When testing for a particular response from the stub, you don't need to use `with()` constraints as the focus is on what we do after that response rather than in what we pass. The `at()` constraints are enough for the stub to discriminate between the first and the second call.